spec.avsc

```diff
@@ -1,22 +1,56 @@
 {
-  "namespace":"io.cloudevents",
+  "namespace": "io.cloudevents.avro",
```

Comment:
Really nice!

Reply:
we would need for sure probably a schema for 0.3 , the `schemaurl` would need to have a default of `null` like the 1.0 for `dataschema`.  this will ensure forward and backward compatiblity. Changing the namespace to `io.cloudevents.v03.avro` also would be needed i think

The most confusing part is indeed the implementation with SDK's, since an avro schema can be used to generate code in multiple language, the need of the java-sdk is not that useful, because  you could use the Generated CloudEvent for validation with avro packages for serialization. 

adding format/avro  could however  be one of these: 
- implementing a serialization that take sdk's cloudevent, converting it to avro cloudevent ( without validation ) then serializing it with avro libraries.  This conversion is necessary since we cannot put directly the avroCloudEvent in the EventFormat class because it takes sdk's cloudevent as an argument. The serializer would also need to do version handling like the `formats/json` do 

However as a developer It is tempting to just use the Avro CloudEvents directly since it provides all the necessary features without the EventFormat overhead

here's an example kafka Serializer for avro  for guidance 
```
import org.apache.avro.generic.GenericDatumWriter;
import org.apache.avro.generic.GenericRecord;
import org.apache.avro.io.BinaryEncoder;
import org.apache.avro.io.DatumWriter;
import org.apache.avro.io.EncoderFactory;
import org.apache.avro.specific.SpecificRecordBase;
import org.apache.kafka.common.errors.SerializationException;
import org.apache.kafka.common.serialization.Serializer;


import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.util.Map;

public class AvroSerializer<T extends SpecificRecordBase> implements Serializer<T> {

    @Override
    public void close() {
        // No-op
    }

    @Override
    public void configure(Map<String, ?> arg0, boolean arg1) {
        // No-op
    }

    @Override
    public byte[] serialize(String topic, T data) {
        try {
            byte[] result = null;

            if (data != null) {
                // here would be the place where we could convert `data` from 
                // an sdk-cloudevent to an avro-cloudevent
                ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
                BinaryEncoder binaryEncoder =
                        EncoderFactory.get().binaryEncoder(byteArrayOutputStream, null);

                DatumWriter<GenericRecord> datumWriter = new GenericDatumWriter<>(data.getSchema());
                datumWriter.write(data, binaryEncoder);

                binaryEncoder.flush();
                byteArrayOutputStream.close();
                result = byteArrayOutputStream.toByteArray();
            }
            return result;
        } catch (IOException ex) {
            throw new SerializationException(
                    "Can't serialize data='" + data + "' for topic='" + topic + "'", ex);
        }
    }
}

```